### PR TITLE
ipa: add user_auth_type attribute

### PR DESCRIPTION
**Pull request by madhuriupadhye**
Add attribute to add supported user authentication for IPA user.

### CODE REVIEW

**Comment by pbrezina** [comment]
Hi, it would be nice to change the commit subject to reflect what attribute are you adding. E.g. `ipa: add user_auth_type attribute`

File: sssd_test_framework/roles/ipa.py [hunk 1]
```diff
@@ -428,6 +429,8 @@ def add(
         :type shell: str | None, optional
         :param require_password_reset: Require password reset on first login, defaults to False
         :type require_password_reset: bool, optional
+        :param user_auth_type: Types of supported user authentication, deafults to None
```
[the remark this answers is not in the document]
Comment (madhuriupadhye) [reply sssd_test_framework/roles/ipa.py]: updated.

File: sssd_test_framework/roles/ipa.py [hunk 2]
```diff
@@ -475,6 +480,8 @@ def modify(
         :type gecos: str | None, optional
         :param shell: Login shell, defaults to None
         :type shell: str | None, optional
+        :param user_auth_type: Types of supported user authentication, deafults to None
```
[the remark this answers is not in the document]
Comment (madhuriupadhye) [reply sssd_test_framework/roles/ipa.py]: updated.

**Comment by madhuriupadhye** [comment]
> Hi, it would be nice to change the commit subject to reflect what attribute are you adding. E.g. `ipa: add user_auth_type attribute`

updated.

**Pull request merged**